scipy/optimize/tests/test_optimize.py

```diff
@@ -2239,6 +2240,42 @@ def test_gh12696():
             lambda x: np.array([x**2]), -np.pi, np.pi, disp=False)
 
 
+eb_cases = ({"fun": optimize.rosen, "jac": False},
+            {"fun": optimize.rosen, "jac": optimize.rosen_der},
+            {"fun": (lambda x: optimize.rosen, optimize.rosen_der),
+             "jac": True})
+
+@pytest.mark.parametrize('method', ['L-BFGS-B', 'SLSQP',
+                                    'trust-constr', 'TNC'])
+@pytest.mark.parametrize('kwds', eb_cases)
+def test_equal_bounds(method, kwds):
```

Comment:
This test is rudimentary. Need to test different combinations of callable/finite difference gradient/constraint jacobian, and we need to look for NaNs in the right places of the output.